tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/QueryFixerMain.java

```diff
@@ -1,227 +1,75 @@
+    if (query == null) {
       // In CLI mode, all the instructions are output by print functions, because logger outputs
       // extract info (time, code position) that distracts users.
       System.out.println(
           "Please provide the query as an argument, enclosed by double quote. Use --help for instruction.");
+      exit(1);
     }
-
-    String query = cmd.getArgList().get(0);
-    AutomaticQueryFixer queryFixer = new AutomaticQueryFixer(bigQueryOptions);
-
-    String mode = cmd.getOptionValue(MODE);
-    if (mode == null) {
-      mode = AUTO_MODE;
-    }
-    switch (mode) {
-      case AUTO_MODE:
-        // TODO: Implement Non-interactive mode
-        FixResult fixResult = queryFixer.fix(query);
-        if (fixResult.getOptions().isEmpty()) {
-          return;
-        }
-        String newQuery = fixResult.getOptions().get(0).getFixedQuery();
-        printQueryResult(newQuery, bigQueryOptions);
-        break;
-
-      case USER_ASSISTANCE_MODE:
-      case UA_MODE:
-        // TODO: Implement guide mode
-        return;
-
-      case FIX_ONCE_MODE:
-      case FO_MODE:
-        fixResult = fixQueryInFullInteractMode(queryFixer, query);
-        printFixResult(query, fixResult, cmd.getOptionValue(OUTPUT));
-        break;
-
-      default:
-        System.out.println("Mode (-m) is incorrect. Use --help for usage.");
-        System.exit(1);
-    }
+    String mode = queryFixerOptions.getOptionValue(MODE);
+    String outputFormat = queryFixerOptions.getOptionValue(OUTPUT);
+    CommandLineInteraction interaction =
+        CommandLineInteraction.create(mode, outputFormat, bigQueryOptions);
+    interaction.interact(query);
   }
 
-  private static BigQueryOptions buildBigQueryOptions(String credentialPath, String projectId) {
-    if (credentialPath == null) {
-      return BigQueryOptions.newBuilder().setProjectId(projectId).build();
-    } else {
-      // TODO: should support this in near future.
-      System.out.println("Customized credential path is not supported");
-      System.exit(1);
-      return null;
-    }
-  }
-
-  private static FixResult fixQueryInFullInteractMode(
-      AutomaticQueryFixer queryFixer, String query) {
-    return queryFixer.fix(query);
-  }
-
-  private static void printFixResult(String query, FixResult fixResult, String outputFormat) {
-    if (outputFormat == null || outputFormat.equalsIgnoreCase(NATURAL_OUTPUT)) {
-      System.out.println("Input query: " + query);
-      printFixResultInCommandLine(fixResult);
-    } else if (outputFormat.equalsIgnoreCase(JSON_OUTPUT)) {
-      printFixResultAsJson(fixResult);
-    } else {
-      System.out.println("Output format (-o) is incorrect. Use --help for usage.");
-      System.exit(1);
+  /** Create the BigQueryOption based on user-input credentials path and project ID. */
+  private static BigQueryOptions buildBigQueryOptions(String credentialsPath, String projectId) {
+    if (projectId == null) {
```

Comment:
Will it be empty? Should we check empty too?

Reply:
If it is empty, it will handled inside BigQueryOptions. It should be treated the same as an incorrect project ID. Therefore, no extra handle needs to be taken here.